### PR TITLE
adding uc to snowflake operator

### PR DESCRIPTION
**Pull request by ckonuganti**
Adding a new plugin to copy data from UC to snowflake and run snowflake queries.

## Description

Added two plugins/operators to run snowflake queries and copy data from unity catalog to snowflake. these plugins helps us run the workloads from one environment instead of switching between dbx and sf. As a future enhancement , want to include support for more credential stores #70 

## Related Issue

#63 

## Motivation and Context

As Enterprises might be using multiple solutions for data warehousing . we have a need to copy data from Unity Catalog to snowflake to support few scenarios. Instead of every team handling this common process in their own unique way, creating this plugin that can copy from UC table to Snowflake Table (Full Load, Incremental) and run queries directly on snowflake.

## How Has This Been Tested?

We tested the code changes manually and documented(will attach screenshots). creating a new issue for automating the test cases #68. 

## Screenshots (if appropriate):
Screenshot 2023-12-08 at 10 14 35 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by asingamaneni — DISMISSED** [review]
@ckonuganti Thanks for your first contribution. This looks good, soon we need to work on generalizing the secrets and tests for these operators

File: brickflow_plugins/databricks/uc_to_snowflake_operator.py [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+            from brickflow import ctx
+
+            self.username = ctx.dbutils.secrets.get(self.secret_scope, "username")
+            self.password = ctx.dbutils.secrets.get(self.secret_scope, "password")
+            self.account = ctx.dbutils.secrets.get(self.secret_scope, "account")
+            self.warehouse = ctx.dbutils.secrets.get(self.secret_scope, "warehouse")
+            self.database = ctx.dbutils.secrets.get(self.secret_scope, "database")
+            self.role = ctx.dbutils.secrets.get(self.secret_scope, "role")
+            # self.authenticator = base64.b64encode(
+            #    ctx.dbutils.secrets.get(self.secret_scope, "authenticator").encode("utf-8")).decode("utf-8") or None
```
Comment (stikkireddy) [inline brickflow_plugins/databricks/uc_to_snowflake_operator.py:269]: dont need commented out code

File: pyproject.toml [hunk 2]
```diff
@@ -46,6 +46,7 @@ python-decouple = "3.8"
 pyyaml = "^6.0"
 requests = ">=2.28.2 <3.0.0"
 # cerberus-python-client = {version = "~2.5.4",  optional = true } # Users might have to manually install cerberus-python-client if required
+snowflake = "^0.3.0"
```
Comment (stikkireddy) [inline pyproject.toml:49]: * remove this
* try catch around the import and throw an error indicating for the user to add this library to the job
* update to the latest version if possible; i believe its 0.5.1

File: pyproject.toml [hunk 3]
```diff
@@ -66,6 +65,7 @@ pytest-mock = "^3.10.0"
 types-PyYAML = "*" # only for development purposes no need to make installation req
 types-requests = ">=2.28.11.16 <3.0.0.0" # only for development purposes no need to make installation req
 apache-airflow = "^2.7.3"
+snowflake = "^0.5.1"
```
Comment (stikkireddy) [inline pyproject.toml:68]: hey the import error says to install snowflake-python-connector not snowflake. They are two different libraries. This shouldnt be a full req for brickflow but just for the operator.

Comment (stikkireddy) [reply pyproject.toml:68]: It should only be required as a library to run the workflow in bricks, not needed to be install locally

File: brickflow_plugins/databricks/uc_to_snowflake_operator.py [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
-import os
-from http import HTTPStatus
 
-from brickflow.context import ctx
+try:
+    import snowflake.connector
+except ImportError:
+    raise ImportError(
+        "You must install snowflake library to use run snowflake plugins, "
+        "please try pip install snowflake-connector-python"
```
Comment (stikkireddy) [inline brickflow_plugins/databricks/uc_to_snowflake_operator.py:8]: this should only be required on the cluster that is using this operator. 

@wf.task(libraries="...")

**Review by asingamaneni — APPROVED** [review]
LGTM

**Pull request merged**